clang/include/clang/3C/Constraints.h

```diff
@@ -277,9 +277,14 @@ class Constraint {
   virtual bool operator==(const Constraint &Other) const = 0;
   virtual bool operator!=(const Constraint &Other) const = 0;
   virtual bool operator<(const Constraint &Other) const = 0;
-  virtual std::string getReason() { return REASON; }
+  virtual std::string getReason() const { return REASON; }
   virtual void setReason(const std::string &Rsn) { REASON = Rsn; }
 
+  bool isUnwritable(void) const {
+    return getReason() == "Declaration in non-writable file" ||
+            getReason() == "Expression in non-writable file";
```

Comment:
Is there a way we can avoid introducing a bug where we add another "X in non-writable file" reason somewhere in 3C and forget to update this function?  Maybe look for "non-writable file" as a substring?  @john-h-kastner Do you have a better idea?

Reply:
I mean we could add a field that marks a CV as having a non-writable reason for wildness?

Reply:
The `ConstraintVariable` or the `Constraint`?  A field on the `Constraint` would be a clean and direct solution to the problem; it would add some memory usage, but the existing `std::string REASON` field may be a bigger offender.  But I think we should get an opinion from John on the best design for this.  Another possibility is to have a numeric priority field.

Reply:
Yes on the constraint.

Reply:
I agree that some sort of indicator on the constraint is better than comparing the reason string. I'm undecided on if a numeric priority or boolean flag is better overall, but I don't think it's terribly important since we hope to eventually support multiple constraints per atom, after which we could remove this new field. I think I lean slightly towards using a flag.  

Regarding the reason strings, it probably wouldn't be hard to intern the reason strings to avoid storing a copy of one of the same few reasons on every constraint.

Reply:
I agree with John on going with the boolean flag as opposed to a counter since this is intended to be a temporary fix

Reply:
Aaron and I discussed this in person, and decided that adding a flag onto the Constraint class was not actually a good idea. The flag would need to be threaded through multiple levels of function calls from multiple call sites, and any future constraints added for unwritable code would need to remember to set the flag.

The alternative we came up with was to choose a single reason string used to represent any Constriant whose cause it due to an unwritable file, `#define` that in a header, and use that in the comparison and as the reason string when the unwritable file constraint is added.

Reply:
This seems like a reasonable solution.  Thanks.